src/pybind/mgr/dashboard_v2/models/nodb.py

```diff
@@ -0,0 +1,892 @@
+            return False
+        return attr not in self.__dict__
+
+    def set_read_only_fields(self, obj, include_pk=True):
+        """
+        .. example::
+            >>> insert = self.id is None
+            >>> diff, original = self.get_modified_fields(name=self.name) if insert
+            >>>     else self.get_modified_fields()
+            >>> if not insert:
+            >>>     self.set_read_only_fields()
+        """
+        if include_pk:
+            self.pk = obj.pk
+
+        for field in self.__class__._meta.fields:
+            if (not field.editable
+               and not self.attribute_is_unevaluated_lazy_property(field.attname)
+               and hasattr(obj, field.attname)
+               and getattr(self, field.attname, None) != getattr(obj, field.attname)):
+                setattr(self, field.attname, getattr(obj, field.attname))
+
+    @classmethod
+    def make_model_args(cls, json_result, fields_force_none=None):
+        """
+        TODO: fields_force_none could be auto generated by the field names.
+
+        :type json_result: dict[str, Any]
+        :type fields_force_none: list[str]
+        :rtype: dict[str, Any]
+        """
+        def get_val_from_json(key):
+            if key in json_result:
+                return json_result[key]
+            elif key.replace('_', '-') in json_result:
+                # '-' is not supported for field names, but used by ceph.
+                return json_result[key.replace('_', '-')]
+            raise AttributeError
+
+        def handle_field(field):
+            """:rtype: list[tuple[str, Any]]"""
+            try:
+                val = get_val_from_json(field.attname)
+            except AttributeError:
+                return []
+
+            if val is None and not field.null:
+                return []
+
+            try:
+                python_val = field.to_python(val)
+            except ValidationError as e:
+                return []
+
+            return [(field.attname, python_val)]
+
+        model_args = dict(
+            itertools.chain.from_iterable([handle_field(field) for field in cls._meta.fields])
+        )
+        for name in fields_force_none or []:
+            if name not in model_args:
+                model_args[name] = None
+        return model_args
+
+    def __init__(self, *args, **kwargs):
+        self._state = mock.Mock()
+
+        for k, w in kwargs.items():
+            self.__dict__[k] = w  # __dict__ is a dictproxy
+
+        for field in self._meta.concrete_fields:
+            # set defaults:
+            if not self.attribute_is_unevaluated_lazy_property(field.name) \
+               and not hasattr(self, field.name):
+                setattr(self, field.name, field.get_default())
+
+    def save(self, force_insert=False, force_update=False, using=None,
+             update_fields=None):
+        """
+        This base implementation does nothing, except telling django that self is now successfully
```

Comment:
Just a general question: I read 'django' serveral times now. Should we remove it on long term?

Reply:
It makes sense to review all comments at a later stage, yes.